### PR TITLE
Respect the "private" flag in .torrent files

### LINKED ISSUE #271: Respect the "private" flag in .torrent files
When torrents are marked "private", they should not announce to any source but the tracker (i.e. no DHT, no peer exchange).

### PR DESCRIPTION
**Pull request by gillesdemey**
Fixes #271 

Couple of caveats:

* The DHT client still throws the ready event, even though no-one is listening for it. This could be patched in `bittorrent-dht` to check if the server hasn't been destroyed yet before we emit the event.

* No tests to check if PEX is correctly disabled.

### CODE REVIEW

File: lib/torrent.js [hunk 1]
```diff
@@ -207,6 +207,12 @@ Torrent.prototype._onParsedTorrent = function (parsedTorrent) {
     })
   }
 
+  // destroy DHT client if the torrent is private
+  if (self.parsedTorrent.private && self.client.dht) {
+    self.client.dht.destroy()
+    self.client.dht = null
+  }
```
Comment (feross) [inline lib/torrent.js]: Other torrents might not be private, and we want the DHT to be available to them. Best to solve this a different way that preserves the DHT for use by other torrents.

Comment (gillesdemey) [inline lib/torrent.js]: Oops, yeah that doesn't seem right.

Then disabling DHT for the discovery and setting DHT to false when passing the handshake options to bittorrent-protocol should suffice?

I'll take another stab at this.

**Comment by feross** [comment]
Excellent PR!

Unfortunately, lots of WebTorrent code changed recently as part of efforts to make WebTorrent and torrent-stream/peerflix share more code in common. Would you mind rebasing these changes so they merge cleanly?

**Comment by gillesdemey** [comment]
I'll take another look, I had noticed that `parsedTorrent` had been removed.

**Comment by feross** [comment]
Right. Instead of doing `torrent.parsedTorrent`, all those properties now exist directly on the `torrent` object.

**Comment by gillesdemey** [comment]
Rebased and updated to reflect the changes in master

**Comment by feross** [comment]
Nice work! Released as 0.62.0.

I also added you as a GitHub collaborator. Feel free to contribute as you see fit :)

**Comment by feross** [comment]
Scratch that, hehe. I already added you from a previous PR :+1:

**Pull request merged**